langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/internal/client/DefaultAnthropicClient.java

```diff
@@ -132,6 +132,9 @@ public AnthropicCreateMessageResponse createMessage(AnthropicCreateMessageReques
     }
 
     private String toBeta(AnthropicCreateMessageRequest request) {
+        if (beta != null) {
```

Comment:
Please double-check this change, it override existing logic

Reply:
I changed it so that it uses the logic to validate whether it has tools or whether it has the beta header filled in.

We can remove the hasTools logic, this feature is no longer in beta, there is no need for the tools-2024-04-04 header

Reply:
A good solution is to leave only the validation if the beta header is filled in, which meets all future beta scenarios.

Reply:
Oh, good point regarding  `tools-2024-04-04` being redundant, let's remove it then. I have tested it locally and pushed the change, I hope you don't mind

Reply:
> A good solution is to leave only the validation if the beta header is filled in, which meets all future beta features.

Which validation do you mean?

Reply:
Validate if the beta field is not null, or leave it as it is now because it did not present any errors.
![Screenshot from 2024-11-07 19-31-17](https://github.com/user-attachments/assets/b07c7506-0cfe-4797-8f30-1cc850601d77)